mine-diamonds/state-transition/src/stf/v1/persist/global.ts

```diff
@@ -0,0 +1,171 @@
+import type { SQLUpdate } from '@paima/node-sdk/db';
+import type Prando from '@paima/sdk/prando';
+import type { WalletAddress } from '@paima/sdk/utils';
+import type {
+  ICancelDexOrderParams,
+  ICreateAssetOwnershipParams,
+  ICreateAssetTokenStateParams,
+  ICreateDexOrderParams,
+  ICreateGlobalUserStateParams,
+  ICreateUserTokenStateParams,
+  IUpdateAssetOwnershipParams,
+  IUpdateDexOrderParams,
+  IUpdateUserStateCurrentTokenIdParams,
+} from '@game/db';
+import {
+  cancelDexOrder,
+  createAssetOwnership,
+  createAssetTokenState,
+  createDexOrder,
+  createGlobalUserState,
+  updateAssetOwnership,
+  updateDexOrder,
+} from '@game/db';
+import { createUserTokenState } from '@game/db';
+import { updateUserStateCurrentTokenId } from '@game/db';
+import type {
+  AssetMintedInput,
+  AssetTransferredInput,
+  OrderCancelledInput,
+  OrderCreatedInput,
+  OrderFilledInput,
+} from '../types';
+
+export function submitMineAttempt(player: WalletAddress, randomnessGenerator: Prando): SQLUpdate[] {
+  return [
+    persistNewUser(player),
+    persistMineAttempt(
+      player,
+      randomnessGenerator.nextInt(1, 10),
+      randomnessGenerator.nextInt(1, 10) <= 7
+    ),
+    persistUpdateUserStateCurrentTokenId(player),
+  ];
+}
+
+export function orderCreated(inputData: OrderCreatedInput): SQLUpdate[] {
+  return [
+    persistOrderCreated(
+      inputData.payload.seller.toLowerCase(),
+      Number(inputData.payload.assetAmount),
+      Number(inputData.payload.assetId),
+      Number(inputData.payload.orderId),
```

Comment:
@matejos Is there a reason why we let the user/frontend send the OrderID? 
I see you have a `ON CONFLICT(orderId) DO NOTHING;`

Reply:
Instead of using, for example a serial id

Reply:
I said nothing, this comes from the orderbook contract events